acceptance/framework/config/config.go

```diff
@@ -22,16 +22,67 @@ const (
 	LicenseSecretKey  = "key"
 )
 
-// TestConfig holds configuration for the test suite.
-type TestConfig struct {
-	Kubeconfig    string
+type KubeEnv struct {
+	KubeConfig    string
 	KubeContext   string
 	KubeNamespace string
+}
+
+func KubeEnvListFromStringList(kubeConfigs, kubeContexts, kubeNamespaces []string) []KubeEnv {
```

Comment:
❓ why don't we just validate that the lists are the same length because otherwise it will get ambiguous. 

So if I want to say `-kube-context=kind-dc1 -kube-context=kind-dc2 -namespace=consul`, we can only interpret it as:
- the first kube-context has the namespace `consul`
- the second kube-context has default namespace of that context

I think enforcing the same length could be simpler because I'm not sure we need this complexity for our use cases anyways. 

So I'm suggesting:
- If you've provided any 2 of the three, their lengths should be equal
- If you've provided all 3, their lengths should be equal as well

Then we just construct the env from those arrays by their position as we don't need to figure out the one with max len.

Reply:
So basically, if you provide any of the flags `kubecontext` `kubeconfig` and `namespaces` then we enforce theya are the same length. If a flag is not provided/list is empty then we just set it all to empty so it's processed as before?

Here's are examples from what I'm understanding:
```
-kubecontexts="kind-dc1,kind-dc2" -namespaces="consul" # would result in an error
```
```
-kubecontexts="kind-dc1,kind-dc2" -namesapces="consul,consul2" # same length so OK!
# Result Structs (we don't set `kube-config` since it wasn't provided)
[]KubeEnv{
  {
    KubeContext: "kind-dc1"", 
    KubeNamespace: "consul"
  },
  {
    KubeContext: "kind-dc2"", 
    KubeNamespace: "consul2"
  },
}
```

Reply:
I think we still need to find out the max length so we can pad the one that's 0 if one of them isn't set?

Reply:
>If a flag is not provided/list is empty then we just set it all to empty so it's processed as before?

Yeah, exactly.

Yep your examples are exactly what I meant.

Reply:
> I think we still need to find out the max length so we can pad the one that's "" if one of them isn't set?

could we rely on defaults in KubeEnv instead? If length is 0 for namespaces for example, then kubeenv for that index will end up not setting the namespace field which should be the same as `""`

Reply:
I think no matter how you slice it if you're constructing a new list you'd still have to get the the length of the new list to create, so you're right we don't technically need to pad, we could check to see if the list is zero and if it not index through it... I'm not sure if that's cleaner?

Reply:
I guess we could do something like this:
```
	out := make([]KubeTestConfig, 0)
	if len(kubeConfigs) == 0 && len(kubeContexts) == 0 && len(kubeNamespaces) == 0{
		out = append(out, KubeTestConfig{})
		return out
	}
	
	i := 0
	for true{
		kenv := KubeTestConfig{}
		if i != 0 && (i == len(kubeConfigs) || i == len(kubeContexts) || i == len(kubeNamespaces)){
			break
		}
		if len(kubeConfigs) != 0{
			kenv.KubeConfig = kubeConfigs[i]
		}
		if len(kubeContexts) != 0{
			kenv.KubeContext = kubeContexts[i]
		}
		if len(kubeNamespaces) != 0{
			kenv.KubeNamespace = kubeNamespaces[i]
		}
		out = append(out, kenv)
	}
	return out
```

Reply:
That's true. You could do something like this which is less code but I dunno if it's that much simpler because you're right you still need to check the length:

```
       l := math.Max(
		float64(len(kubeConfigs)),
		math.Max(float64(len(kubeContexts)), float64(len(kubeNamespaces))))

	// If all are empty, then return a single empty entry
	if l == 0 {
		return []KubeEnv{{}}
	}

	out := make([]KubeEnv, int(l))

	for i := range out {
		if len(kubeConfigs) != 0 {
			out[i].KubeConfig = kubeConfigs[i]
		}

		if len(kubeContexts) != 0 {
			out[i].KubeContext = kubeContexts[i]
		}

		if len(kubeNamespaces) != 0 {
			out[i].KubeNamespace = kubeNamespaces[i]
		}
	}

	return out
```

Reply:
oh yeah, `math.Max` is a nice call

Reply:
The only thing I changed was checking for `> i` instead of `!=0`

This guarantees we don't index out of range, although we are now validating that lengths are either 0 or equal